### PR TITLE
Mathew131 plots

### PR DESCRIPTION
**Pull request by Mathew131**

### CODE REVIEW

File: main/mainwindow.cpp [hunk 1]
```diff
@@ -1,86 +1,114 @@
...
-  ui->tableViewNaming->horizontalHeader()->setSectionResizeMode(
-      QHeaderView::ResizeToContents);
-
-  ui->tableViewMain->show();
-  ui->tableViewNaming->show();
-
-  connect(ui->addRowBtn, SIGNAL(clicked()), this, SLOT(addRow()));
-  connect(ui->addColumnBtn, SIGNAL(clicked()), this, SLOT(addColumn()));
-  connect(ui->deleteRowBtn, SIGNAL(clicked()), this, SLOT(removeRow()));
-  connect(ui->deleteColumnBtn, SIGNAL(clicked()), this, SLOT(removeColumn()));
```
Comment (TheFueRr) [inline main/mainwindow.cpp]: clang format otdihaet

File: main/mainwindow.cpp [hunk 2]
```diff
@@ -1,86 +1,114 @@
...
+    ui->tableViewNaming->horizontalHeader()->setSectionResizeMode(
+        QHeaderView::ResizeToContents);
+
+    ui->tableViewMain->show();
+    ui->tableViewNaming->show();
+
+    connect(ui->addRowBtn, SIGNAL(clicked()), this, SLOT(addRow()));
+    connect(ui->addColumnBtn, SIGNAL(clicked()), this, SLOT(addColumn()));
+    connect(ui->deleteRowBtn, SIGNAL(clicked()), this, SLOT(removeRow()));
+    connect(ui->deleteColumnBtn, SIGNAL(clicked()), this, SLOT(removeColumn()));
```
Comment (TheFueRr) [inline main/mainwindow.cpp]: clang format otdihaet 1

File: main/mainwindow.cpp [hunk 3]
```diff
@@ -1,86 +1,114 @@
...
 }
 
 MainWindow::~MainWindow() { delete ui; }
 
 void MainWindow::on_deletePlotBtn_clicked() {
-  int index = ui->tabWidgetPlots->currentIndex();
-  ui->tabWidgetPlots->removeTab(index);
+    int index = ui->tabWidgetPlots->currentIndex();
+    ui->tabWidgetPlots->removeTab(index);
```
Comment (TheFueRr) [inline main/mainwindow.cpp]: clang format otdihaet 2

File: main/mainwindow.cpp [hunk 4]
```diff
@@ -1,86 +1,114 @@
...
 }
 
 void MainWindow::on_addPlotBtn_clicked() {
-  int count = ui->tabWidgetPlots->count();
-  ui->tabWidgetPlots->addTab(new QCustomPlot,
-                             "Plot " + QString::number(count + 1));
+    int count = ui->tabWidgetPlots->count();
+    ui->tabWidgetPlots->addTab(new QCustomPlot,
+                               "Plot " + QString::number(count + 1));
```
Comment (TheFueRr) [inline main/mainwindow.cpp]: clang format otdihaet 3

File: main/mainwindow.cpp [hunk 5]
```diff
@@ -1,86 +1,114 @@
...
-      lib::Manager::getInstance()->getVariable(i).measurements.push_back(0);
-  ui->tableViewMain->model()->insertRows(
-      lib::Manager::getInstance()->getMeasurementsCount(), 1);
+    size_t count = lib::Manager::getInstance()->getMeasurementsCount();
+    for (int i = 0; i < lib::Manager::getInstance()->getVariablesCount(); i++)
+        if (count ==
+            lib::Manager::getInstance()->getVariable(i).getMeasurementsCount())
+            lib::Manager::getInstance()->getVariable(i).measurements.push_back(0);
+    ui->tableViewMain->model()->insertRows(
+        lib::Manager::getInstance()->getMeasurementsCount(), 1);
```
Comment (TheFueRr) [inline main/mainwindow.cpp]: clang format otdihaet 4

File: main/mainwindow.cpp [hunk 6]
```diff
@@ -1,86 +1,114 @@
...
-  ui->tableViewMain->model()->removeRows(
-      lib::Manager::getInstance()->getMeasurementsCount(), 1);
+    size_t count = lib::Manager::getInstance()->getMeasurementsCount();
+    if (count == 1) return;
+    for (int i = 0; i < lib::Manager::getInstance()->getVariablesCount(); i++)
+        if (count ==
+            lib::Manager::getInstance()->getVariable(i).getMeasurementsCount())
+            lib::Manager::getInstance()->getVariable(i).measurements.pop_back();
+    ui->tableViewMain->model()->removeRows(
+        lib::Manager::getInstance()->getMeasurementsCount(), 1);
```
Comment (TheFueRr) [inline main/mainwindow.cpp]: clang format otdihaet 5

File: main/mainwindow.cpp [hunk 7]
```diff
@@ -1,86 +1,114 @@
...
-  ui->tableViewMain->model()->removeColumns(
-      lib::Manager::getInstance()->getVariablesCount(), 1);
-  ui->tableViewNaming->model()->removeRows(
-      lib::Manager::getInstance()->getVariablesCount(), 1);
+    if (lib::Manager::getInstance()->getVariablesCount() == 1) return;
+    lib::Manager::getInstance()->deleteVariable();
+    ui->tableViewMain->model()->removeColumns(
+        lib::Manager::getInstance()->getVariablesCount(), 1);
+    ui->tableViewNaming->model()->removeRows(
+        lib::Manager::getInstance()->getVariablesCount(), 1);
```
Comment (TheFueRr) [inline main/mainwindow.cpp]: clang format otdihaet 6

File: main/mainwindow.cpp [hunk 8]
```diff
@@ -1,86 +1,114 @@
...
-  lib::Manager::getInstance()->addVariable(lib::Variable());
-  ui->tableViewMain->model()->insertColumns(
-      lib::Manager::getInstance()->getVariablesCount(), 1);
-  ui->tableViewNaming->model()->insertRows(
-      lib::Manager::getInstance()->getVariablesCount(), 1);
+    lib::Manager::getInstance()->addVariable(lib::Variable());
+    ui->tableViewMain->model()->insertColumns(
+        lib::Manager::getInstance()->getVariablesCount(), 1);
+    ui->tableViewNaming->model()->insertRows(
+        lib::Manager::getInstance()->getVariablesCount(), 1);
```
Comment (TheFueRr) [inline main/mainwindow.cpp]: clang format otdihaet 7

File: main/mainwindow.cpp [hunk 9]
```diff
@@ -1,86 +1,114 @@
...
+        QVector<double> a;
+        for (int j = 0; j < (lib::Manager::getInstance()->getVariableMeasurements(i)).size(); ++j) {
+            if ((lib::Manager::getInstance()->getVariableMeasurements(i))[j])
+                a.push_back((lib::Manager::getInstance()->getVariableMeasurements(i))[j]);
+        }
+        graph->setData({1, 2, 3, 4, 5}, a);
+    }
+}
+
+MainWindow::MainWindow(QWidget* parent)
```
Comment (TheFueRr) [inline main/mainwindow.cpp]: shifting the constructor of the mainwindow class is a bad idea

**Review by sh1shk1nsky — APPROVED** [review]
Need to add discription for several names

**Pull request merged**